pkg/certificate/certificate.go

```diff
@@ -57,9 +57,15 @@ func ValidateConnectionContext(ctx context.Context, pool *x509.CertPool, targetU
 				}
 
 				cert := state.PeerCertificates[0]
+				intermediates := x509.NewCertPool()
+				for _, intermediate := range state.PeerCertificates[1:] {
```

Comment:
can we run into an out of range err here?

Reply:
Thanks for checking!

It seems that at this point we have `len(state.PeerCertificates) > 0`, so with a slice of 1 this is a-ok (won't loop)

Reply:
ah right, that will produce an empty slice if len()==1